src/client/executor.rs

```diff
@@ -317,7 +323,10 @@ impl Client {
         }
 
         self.emit_command_event(|handler| {
-            let should_redact = REDACTED_COMMANDS.contains(cmd.name.to_lowercase().as_str());
+            let name = cmd.name.to_lowercase();
+            let should_redact = REDACTED_COMMANDS.contains(name.as_str())
+                || HELLO_COMMAND_NAMES.contains(name.as_str())
+                    && cmd.body.contains_key("speculativeAuthenticate");
```

Comment:
We were missing redaction for the last command in this table https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#security